src/Avalonia.Visuals/Rendering/ICustomSimpleHitTest.cs

```diff
@@ -0,0 +1,12 @@
+namespace Avalonia.Rendering
+{
+    /// <summary>
+    /// An interface to allow non-templated controls to customize their hit-testing
+    /// when using a renderer with a simple hit-testing algorithm without a scene graph,
+    /// such as <see cref="ImmediateRenderer" />
+    /// </summary>
+    public interface ICustomSimpleHitTest
```

Comment:
My preference would be for something like `IImmediateHitTest` to be honest - it'd be clearer what it's for. Do you foresee other renderers that will use this interface?

Reply:
I can forsee this being used for any renderer that doesn't use the scene graph and wants to use a simple hit testing algorithm based off control bounds instead of something that inspects draw bounds and transparency. I can change the name though if you want.

Reply:
I'd prefer something like `ImmediateHitTest` but I'll let you decide, it's your PR ;)